splink/postgres/postgres_helpers/postgres_base.py

```diff
@@ -0,0 +1,64 @@
+from ...dialect_base import DialectBase
+
+
+def size_array_intersect_sql(col_name_l, col_name_r):
+    return f"cardinality(array_intersect({col_name_l}, {col_name_r}))"
+
+
+def datediff_sql(
+    col_name_l,
+    col_name_r,
+    date_threshold,
+    date_metric,
+    cast_str=False,
+    date_format=None,
+):
+    if date_metric not in ("day", "month", "year"):
+        raise ValueError("`date_metric` must be one of ('day', 'month', 'year')")
+
+    if date_format is None:
+        date_format = "yyyy-MM-dd"
+
+    if cast_str:
+        datediff_args = f"""
+            to_date({col_name_l}, '{date_format}'),
+            to_date({col_name_r}, '{date_format}')
+        """
+    else:
+        datediff_args = f"{col_name_l}, {col_name_r}"
+
+    if date_metric == "day":
+        date_f = f"""
+            abs(
+                datediff(
+                    {datediff_args}
+                )
+            )
+        """
+    elif date_metric in ["month", "year"]:
+        date_f = f"""
+            floor(abs(
+                ave_months_between(
+                    {datediff_args}
+                )"""
+        if date_metric == "year":
+            date_f += " / 12))"
+        else:
+            date_f += "))"
+    return f"""
+        {date_f} <= {date_threshold}
+    """
+
+
```

Comment:
Can you add a function for regex extraction similar to [here](https://github.com/moj-analytical-services/splink/blob/d6e3ea1e48f9501617d40a581629de4d7c4e4383/splink/duckdb/duckdb_helpers/duckdb_base.py#L38)? That way we could make `postcode_comparison` from `ctl` available for postgres

Reply:
If it's alright I might save that for a separate PR - already some functionality that's not included in this and don't want to make this any more massive than it already is. Will raise issues for all the outstanding missing features/ functionality

Reply:
Yep totally makes sense 👍

Reply:
FYI: #1330